src/SQONBuilder.ts

```diff
@@ -36,18 +125,135 @@ type SQONBuilder = {
 	const toString = () => JSON.stringify(_sqon);
 	const toValue = () => _sqon;
+
+	/* ===== Filter modification ===== */
+	const removeExactFilter = (filter: FilterOperator): SQONBuilder => {
+		if (isFilter(_sqon)) {
+			if (checkMatchingFilter(_sqon, filter)) {
+				// The entire sqon matches the filter to remove, so we will return an empty and combination operator
+				return createBuilder(emptySQON());
+			} else {
+				return createBuilder(_sqon);
+			}
+		} else {
+			const filteredContent = _sqon.content.filter(
+				(operator) => !isFilter(operator) || !checkMatchingFilter(operator, filter),
+			);
+			const updated: CombinationOperator = { op: _sqon.op, content: filteredContent };
+			return createBuilder(updated);
+		}
+	};
+
+	const removeFilter = (fieldName: string, op?: FilterKey, value?: FilterValue): SQONBuilder => {
+		// To prevent repeatedly converting value into an array, do it once at the top
+		// This is now our list of values we want to filter:
+		const valuesToFilter = value !== undefined ? asArray(value) : undefined;
+
+		/* ===== Private Functions for Matching ===== */
+		// Conditional matching based on all the provided arguments
+		function isMatchArgs(filter: FilterOperator) {
+			return (
+				fieldName === filter.content.fieldName &&
+				(op === undefined || op === filter.op) &&
+				(valuesToFilter === undefined || checkMatchingArrays(valuesToFilter, asArray(filter.content.value)))
+			);
+		}
+
+		// Conditional matching based on fiedlName and op only - use this to find partial matchs on array filters before filtering array values
+		function isPartialMatchArgs(filter: FilterOperator) {
+			return fieldName === filter.content.fieldName && (op === undefined || op === filter.op);
+		}
+
+		// Clones filter and then removes values from content.value
+		function filterValuesFromFilter(filter: ArrayFilter, valuesToFilter: (string | number)[]): ArrayFilter {
+			const output = cloneDeepValues(filter);
+			output.content.value = asArray(output.content.value).filter((value) => !valuesToFilter.includes(value));
+			return output;
+		}
+
+		/* ===== Remove Filter Work ===== */
+		// First check if the entire sqon is a filter, if so we can apply the matching logic to just that opeartor
+		if (isFilter(_sqon)) {
+			if (isMatchArgs(_sqon)) {
+				// Remove entire sqon since it matches the filter args
+				return createBuilder(emptySQON());
+			} else if (valuesToFilter !== undefined && isArrayFilter(_sqon) && isPartialMatchArgs(_sqon)) {
+				// Filter wasnt removed, but it matches partially so we can filter out specific filter values
+				const output = filterValuesFromFilter(_sqon, valuesToFilter);
+				return createBuilder(output);
+			} else {
+				// No match, no change
+				return createBuilder(_sqon);
+			}
+		}
+
+		// Filter content to remove all exact matches
+		const filteredContent = _sqon.content.filter((operator) => isCombination(operator) || !isMatchArgs(operator));
+
+		if (valuesToFilter !== undefined) {
```

Comment:
avoidable negative logic

Reply:
You would have probably accepted `if (valuesToFilter)` as the clause. This is conceptually equivalent (but allows empty strings).

Reply:
Actually, empty strings are falsy, so those wouldn't pass either.